react-app/src/SignUp.js

```diff
@@ -1,54 +1,100 @@
 import React from "react";
 import Slider from "@material-ui/core/Slider";
 
+const axios = require("axios").default;
+const fs = require("fs");
+const jwt = require("jsonwebtoken");
+
+
+// generic axios request
+function axiosRequest(request) {
+  console.log(request);
+  axios[request.type](request.url, request.data)
+    .then(function(response) {
+      console.log(response);
+      if (response.status === 200) {
+        request.onSuccess(response);
+      }
+    })
+    .catch(error => console.log(error));
+}
+
 class SignUp extends React.Component {
   constructor(props) {
     super(props);
 
     this.defaultPriceRange = [10, 20];
 
     this.state = {
-      username: null,
+      userid: null,
       nickname: null,
-      minPrice: this.defaultPriceRange[0],
-      maxPrice: this.defaultPriceRange[1],
+      priceMin: this.defaultPriceRange[0],
+      priceMax: this.defaultPriceRange[1],
       picture: null,
       formState: 0
     };
 
     this.handleNicknameChange = this.handleNicknameChange.bind(this);
     this.handlePriceRangeChange = this.handlePriceRangeChange.bind(this);
+    this.handlePictureChange = this.handlePictureChange.bind(this);
     this.submitUserSetup = this.submitUserSetup.bind(this);
     this.nextForm = this.nextForm.bind(this);
     this.prevForm = this.prevForm.bind(this);
 
     // load user
-    var hashObject = {}
-    for (let info of window.location.hash.split("&")) {
-      let key_val = info.split("=")
-      hashObject[key_val[0]] = key_val[1].substring(1)
+    var hashObject = {};
```

Comment:
Check what I wrote! I've made this a lot more stable and got rid of the stupid problem with the hash

Reply:
``` javascript

let hashObject = {};
let str = window.location.href.substring(window.location.href.indexOf('#')+1);
for (let info of str.split("&")) {
 	let stuf = info.split("=");
        hashObject[stuf[0]] = stuf[1];
}


function setCookie(cname, cvalue, exseconds) {
  var d = new Date();
  d.setTime(d.getTime() + (exseconds*1000));
  var expires = "expires="+ d.toUTCString();
  document.cookie = cname + "=" + cvalue + ";" + expires + ";path=/";
}

function getCookie(cname) {
  var name = cname + "=";
  var decodedCookie = decodeURIComponent(document.cookie);
  var ca = decodedCookie.split(';');
  for(var i = 0; i <ca.length; i++) {
    var c = ca[i];
    while (c.charAt(0) == ' ') {
      c = c.substring(1);
    }
    if (c.indexOf(name) == 0) {
      return c.substring(name.length, c.length);
    }
  }
  return "";
}

setCookie("id_token", hashObject["id_token"], hashObject["expires_in"]);
setCookie("access_token", hashObject["access_token"], hashObject["expires_in"]);

```